src/System.Resources.Extensions/src/System/Resources/Extensions/TypeNameComparer.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more informanullable enable
+
+#nullable enable
+using System.Collections.Generic;
+using System.Numerics.Hashing;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+
+namespace System.Resources.Extensions
+{
+    /// <summary>
+    /// Compares type names as strings, ignoring version.
+    /// When type names are missing, mscorlib is assumed.
+    /// This comparer is not meant to capture all scenarios (eg: TypeForwards)
+    /// but is meant to serve as a best effort, avoiding false positives, in the 
+    /// absense of real type meatadata.
+    /// </summary>
+    internal sealed class TypeNameComparer : IEqualityComparer<string>
+    {
+        public static TypeNameComparer Instance { get; } = new TypeNameComparer();
+
+        // these match the set of whitespace characters allowed by the runtime's type parser
+        private static readonly char[] s_whiteSpaceChars =
+        {
+            ' ', '\n', '\r', '\t'
+        };
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static ReadOnlySpan<char> ReadTypeName(ReadOnlySpan<char> assemblyQualifiedTypeName)
+        {
+            // the runtime doesn't tolerate anything between type name and comma
+            int comma = assemblyQualifiedTypeName.IndexOf(',');
+
+            return comma == -1 ? assemblyQualifiedTypeName : assemblyQualifiedTypeName.Slice(0, comma);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static ReadOnlySpan<char> ReadAssemblySimpleName(ReadOnlySpan<char> assemblyName)
+        {
+            int comma = assemblyName.IndexOf(',');
+
+            return comma == -1 ? assemblyName : assemblyName.Slice(0, comma).TrimEnd(s_whiteSpaceChars);
+        }
+
+        private static bool IsMscorlib(ReadOnlySpan<char> assemblyName)
+        {
+            // to match IsMscorlib() in VM, which will ignore public key token and culture for corelib
+            return assemblyName.Equals("mscorlib".AsSpan(), StringComparison.OrdinalIgnoreCase);
+        }
+
+        // Compare two type names ignoring version
+        // If a type name is missing assembly, we assume it came from mscorlib
+        // since this is what Type.GetType will do.
+        public bool Equals(string assemblyQualifiedTypeName1, string assemblyQualifiedTypeName2)
```

Comment:
@BenVillalobos, just making you aware of this. I suspect we're going to want to have a very similar spanified implementation in RAR at some point.

Reply:
Be careful about the mscorlib-isms and the missing name cases here.  Those aren't so general purpose.  Also note that I didn't bother improving the comparison of culture/key that could be done in a non-allocating way if you care about perf.  Also this ignores version which RAR very much cares about.  This is more of a reference and not a reusable bit of code.